---
title: "ED_1_Fzd5_Nanog"
date: "2023-06-06"
output: 
  html_document:
    theme: journal #cerulean
    highlight: monochrome
    toc: true
    toc_float: true
    code_folding: show
  editor_options:
    chunk_output_type: console
---

Scripts for all figures in the paper.
This is Extended Data 1

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Load libraries
```{r}
suppressWarnings(library(GenomicRanges))
suppressWarnings(library(dplyr))
suppressWarnings(library(stringr))
suppressWarnings(library(readr))
suppressWarnings(library(tidyr))
suppressWarnings(library(ggplot2))
suppressWarnings(library(ggplotify))
library(rtracklayer)
library(IRanges)
library(GenomeInfoDb)
library(tidyverse)
library(ggrastr)
library(gridExtra)
library(colorspace)
library(caTools)
library(RColorBrewer)
```


# Extended Data 1A - locus
```{r}
# data
## pA-DamID, CAST allele from my data
DamID <- as_tibble(import("/DATA/usr/l.dauban/projects/pA-DamID/E1970_ProkDNA/2nd_round/results_mm10_nextseq_strainspecific/tracks/normalized/bin-20kb/WT_F1_LB1_CAS-20kb-combined.bw")) %>% dplyr::rename(DamID = score) %>% filter(seqnames == "chr9")

# Function
PlotDataTracksLight <- function(input_tib, exp_names, color_groups, 
                                centromeres = NULL, plot_chr = "chr1", 
                                plot_start = 1, plot_end = 500e6,
                                smooth = 1, color_list = NULL,
                                fix_yaxis = F, aspect_ratio = NULL,
                                lighten_negative = F, raster = F) {
  
  # Get the scores for the samples
  tib_plot <- input_tib %>%
    dplyr::select(seqnames, start, end, 
                  all_of(exp_names))
  
  if (smooth > 1) {
    tib_plot <- tib_plot %>%
      mutate_at(vars(contains("_")), 
                runmean, k = smooth)
  }
  
  # Filter for plotting window
  tib_plot <- tib_plot %>%
    filter(seqnames == plot_chr,
           start >= plot_start,
           end <= plot_end)
  
  # Gather
  tib_gather <- tib_plot %>%
    gather(key, value, -seqnames, -start, -end) %>%
    mutate(key = factor(key, levels = exp_names),
           fill_column = color_groups[match(key, names(color_groups))],
           fill_column = factor(fill_column, levels = unique(color_groups)))
  
  # If desired, make negative values a lighter shade to improve visualization
  if (lighten_negative) {
    tib_gather <- tib_gather %>%
      mutate(fill_column = interaction(fill_column,
                                       value < 0))
  }
  
  
  
  # Plot
  ylimits <- quantile(tib_gather$value, c(0.001, 0.999), na.rm = T)
  fill_column <- NULL
  
  plt <- tib_gather %>% 
    ggplot(aes(fill = fill_column))
  
  
  # Set all counts tracks to the same limits
  if (raster) {
    plt <- plt + 
      rasterize(geom_rect(aes(xmin = start / 1e6, xmax = end / 1e6, 
                              ymin = 0, ymax = value)),
                dpi = 300)
  } else {
    plt <- plt + 
      geom_rect(aes(xmin = start / 1e6, xmax = end / 1e6, 
                    ymin = 0, ymax = value))
  }
  
  plt <- plt + 
    geom_hline(yintercept = 0, size = 0.5) +
    #geom_vline(xintercept = 110927601/1e6, size = 0.5, col = "grey") +
    #geom_vline(xintercept = 111912792/1e6, size = 0.5, col = "grey") +
    #geom_segment(data=clones.df, mapping=aes(x=x/1e6, xend=xend/1e6, y=y, yend=yend), size =1)+
    #geom_segment(data=LAD_loc, mapping=aes(x=x/1e6, xend=xend/1e6, y=y, yend=yend), size =1, color = "#6BAED6")+
    facet_grid(key ~ ., scales = "free_y") +
    xlab(paste0(plot_chr, " (Mb)")) +
    ylab("LaminB1/Dam only") +
    scale_x_continuous(expand = c(0, 0)) + 
    scale_y_continuous(expand = c(0.025, 0.025), limits = c(-4, 1.5)) +
    theme_classic() +
    theme(text = element_text(size = 50))
  
  # Centromeres
  if (! is.null(centromeres)) {
    centromeres.plt <- as_tibble(centromeres) %>%
      filter(seqnames == plot_chr) %>%
      gather(key_centromeres, value, start, end)
    
    plt <- plt +
      geom_line(data = centromeres.plt, 
                aes(x = value / 1e6, y = 0),
                color = "black", size = 2)
  }
  
  if (! is.null(color_list)) {
    
    if (lighten_negative) {
      color_list <- c(color_list,
                      lighten(color_list, amount = 0.5))
    }
    
    colors <- levels(tib_gather$fill_column)
    
    color_list <- color_list[1:length(colors)]
    names(color_list) <- colors
    #colors <- recode(colors, !!!color_list)
    
    plt <- plt +
      scale_fill_manual(values = color_list, guide = "none")
  } else {
    plt <- plt +
      scale_fill_brewer(palette = "Set1", guide = "none")
  }
  
  if (fix_yaxis) {
    plt <- plt + 
      coord_cartesian(xlim = c(max(c(plot_start,
                                     min(tib_gather$start))) / 1e6,
                               min(c(plot_end,
                                     max(tib_gather$end))) / 1e6),
                      ylim = ylimits)
  } else {
    plt <- plt + 
      coord_cartesian(xlim = c(max(c(plot_start,
                                     min(tib_gather$start))) / 1e6,
                               min(c(plot_end,
                                     max(tib_gather$end))) / 1e6))
  }
  
  if (! is.null(aspect_ratio)) {
    plt <- plt +
      theme(aspect.ratio = aspect_ratio)
  }
  
  plot(plt)
  
}

# Plot
PlotDataTracksLight(input_tib = DamID,
                    exp_names = c("DamID"),
                    color_groups = c(DamID = "pA-DamID"),
                    plot_chr = "chr9", plot_start = 107939416, plot_end = 119448158,
                    color_list = c("#A6CEE3", "#F5C374"),
                    #color_list = brewer.pal(3, "Paired"),
                    aspect_ratio = 1/3,
                    smooth = 4,
                    #fix_yaxis = F,
                    lighten_negative = T,
                    raster = T)

# save
# ggsave("SuppFig1A_locus.pdf", path = "./output/", height = 10, width = 20)


```


# Extended Data 1D - SB hopping behaviour in Fzd5 iLAD
```{r}

# Load metadata of integrations
metadata <- dplyr::tibble(file = dir("/DATA/usr/l.dauban/projects/TagMap/E1682_SB_hopped_clones/Output/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::filter(grepl("insertions", file)) %>%
  dplyr::mutate(sample = stringr::str_remove(basename(file), "\\..*"))


# Load all mapped insertion sites into single tibble
tib <- dplyr::bind_rows(lapply(
  1:nrow(metadata),
  function(i) {
    tmp <- readr::read_tsv(metadata$file[i],
      col_types = cols(
        seqnames = col_character(),
        start = col_double(),
        end = col_double(),
        start_gap = col_double(),
        end_gap = col_double(),
        gap_concordance = col_double(),
        count = col_double(),
        D = col_double(),
        p = col_double(),
        concordance = col_double(),
        p_adj = col_double(),
        strand = col_character(),
        name = col_character(),
      )
    ) %>%
      dplyr::mutate(sample = metadata$sample[i])
  }
))

# Convert 'seqnames' to factor
chromosomes <- paste0("chr", c(1:19, "X"))
tib <- tib %>%
  dplyr::mutate(seqnames = factor(seqnames, levels = chromosomes)) %>%
  tidyr::drop_na(seqnames)

# Add factor 'hopped' describing source of hopping
tib <- tib %>%
  dplyr::mutate(sample = factor(sample, )) %>%
  dplyr::mutate(
    hopped = case_when(
      grepl("Fzd5", sample) ~ "Fzd5",
      grepl("Tdgf1", sample) ~ "Tdgf1",
      grepl("Nanog", sample) ~ "Nanog",
      T ~ "unknown"
    ),
    hopped = factor(hopped, c("Fzd5", "Tdgf1", "Nanog"))
  ) %>%
  # Add factor 'library' describing samples originating from pool or clone libraries
  dplyr::mutate(
    library = case_when(
      grepl("_D", sample) ~ "pool",
      grepl("_cl", sample) ~ "clone",
      grepl("_FACS", sample) ~ "pool",
      T ~ "unknown"
    ),
    library = factor(library, c("pool", "clone"))
  )

# data
donor <- GenomicRanges::GRanges(seqnames = "chr1", ranges = IRanges(start = 64986916, end = 64994318), strand = "*")

LAD_left <- data.frame(xmin=60880001/1e6, xmax=62660000/1e6, ymin=-Inf, ymax=Inf)
LAD_right <- data.frame(xmin=65280001/1e6, xmax=66700000/1e6, ymin=-Inf, ymax=Inf)

# Color-blind colors
BLUE <- '#56B4E9'
YELLOW <- '#F0E442'
RED <- '#D55E00'

# process
Fzd5.data <- tib %>%
  filter(hopped == 'Fzd5')
Fzd5.distinct <- Fzd5.data %>%
  select(seqnames, start, end, strand) %>%
  distinct()
Fzd5.gr <- makeGRangesFromDataFrame(Fzd5.distinct, keep.extra.columns = T)
Fzd5.distance <- as_tibble(distanceToNearest(Fzd5.gr, donor, ignore.strand = T)) %>%
  mutate(
    location = start(Fzd5.gr[queryHits]),
    upstream = location < start(donor),
    distance = ifelse(upstream == T, -distance, distance)
  )
Fzd5.distance 
# plot
ggplot(Fzd5.distance, aes(x = location/1e6)) +
  geom_rect(data = LAD_left, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_rect(data = LAD_right, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_histogram(fill = "gray85", color = "gray80",  binwidth = 0.1, alpha = 0.5)+
  geom_density(aes(y=..scaled..*95), n = 4096, size = 1) +
  scale_y_continuous(sec.axis = sec_axis(~./95, name = "Integration density", breaks = seq(0,1,0.2)))+
  geom_rug(length = unit(0.08, "npc")) +
  coord_cartesian(xlim = c(62.2, 66.7), ylim = c(0, 100), expand = FALSE) +
  theme_bw() +
  geom_vline(aes(xintercept = start(donor) / 1e6), col = RED, size = 1) +
  geom_vline(aes(xintercept = LAD_left$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_left$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_right$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_right$xmax), col = BLUE, size = 1) +
  labs(title = 'Fzd5', x = "chr1 (Mb)", y = 'Integration counts') +
  theme(plot.title = element_text(hjust = 0.5),
        text = element_text(size = 50)) 

# save
# ggsave("SuppFig1D_LD20230919_Fzd5.pdf", path = "./output/", height = 10, width = 20)

# number of integration
# total
nrow(Fzd5.distance)

# in the plotted window
Fzd5.distance %>% filter(location > 62.2e6 & location < 66.7e6) %>% nrow()

# number of integrations in iLAD
damid <- import("/DATA/usr/l.dauban/projects/pA-DamID/E1970_ProkDNA/2nd_round/results_mm10_nextseq_strainspecific/tracks/normalized/bin-20kb/WT_F1_LB1_CAS-20kb-combined.bw")

Fzd5_ovl_integ <- findOverlaps(Fzd5.gr, damid, select = "arbitrary")
Fzd5.gr$damid <- damid$score[Fzd5_ovl_integ]
Fzd5.gr$LAD = ifelse(Fzd5.gr$damid>0, "LAD", "iLAD")

Fzd5.gr %>% data.frame() %>% filter(seqnames == "chr1") %>% filter(LAD == "iLAD") %>% nrow()

# number of integrations in iLAD in the plotted window
Fzd5.gr %>% data.frame() %>% filter(seqnames == "chr1") %>% filter(start > 62.2e6 & start < 66.7e6) %>% filter(LAD == "iLAD") %>% nrow()


```


# Extended Data 1E - SB hopping behaviour in Nanog iLAD
```{r}

# Load metadata of integrations
metadata <- dplyr::tibble(file = dir("/DATA/usr/l.dauban/projects/TagMap/E1682_SB_hopped_clones/Output/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::filter(grepl("insertions", file)) %>%
  dplyr::mutate(sample = stringr::str_remove(basename(file), "\\..*"))


# Load all mapped insertion sites into single tibble
tib <- dplyr::bind_rows(lapply(
  1:nrow(metadata),
  function(i) {
    tmp <- readr::read_tsv(metadata$file[i],
      col_types = cols(
        seqnames = col_character(),
        start = col_double(),
        end = col_double(),
        start_gap = col_double(),
        end_gap = col_double(),
        gap_concordance = col_double(),
        count = col_double(),
        D = col_double(),
        p = col_double(),
        concordance = col_double(),
        p_adj = col_double(),
        strand = col_character(),
        name = col_character(),
      )
    ) %>%
      dplyr::mutate(sample = metadata$sample[i])
  }
))

# Convert 'seqnames' to factor
chromosomes <- paste0("chr", c(1:19, "X"))
tib <- tib %>%
  dplyr::mutate(seqnames = factor(seqnames, levels = chromosomes)) %>%
  tidyr::drop_na(seqnames)

# Add factor 'hopped' describing source of hopping
tib <- tib %>%
  dplyr::mutate(sample = factor(sample, )) %>%
  dplyr::mutate(
    hopped = case_when(
      grepl("Fzd5", sample) ~ "Fzd5",
      grepl("Tdgf1", sample) ~ "Tdgf1",
      grepl("Nanog", sample) ~ "Nanog",
      T ~ "unknown"
    ),
    hopped = factor(hopped, c("Fzd5", "Tdgf1", "Nanog"))
  ) %>%
  # Add factor 'library' describing samples originating from pool or clone libraries
  dplyr::mutate(
    library = case_when(
      grepl("_D", sample) ~ "pool",
      grepl("_cl", sample) ~ "clone",
      grepl("_FACS", sample) ~ "pool",
      T ~ "unknown"
    ),
    library = factor(library, c("pool", "clone"))
  )

# data
donor <- GenomicRanges::GRanges(seqnames = "chr6", ranges = IRanges(start = 122650131, end = 122650131), strand = "*")

LAD_left_2 <- data.frame(xmin=121340001/1e6, xmax=121440000/1e6, ymin=-Inf, ymax=Inf)
LAD_left <- data.frame(xmin=121580001/1e6, xmax=122280000/1e6, ymin=-Inf, ymax=Inf)
LAD_right <- data.frame(xmin=122880001/1e6, xmax=124380000/1e6, ymin=-Inf, ymax=Inf)

# Color-blind colors
BLUE <- '#56B4E9'
YELLOW <- '#F0E442'
RED <- '#D55E00'

# process
Nanog.data <- tib %>%
  filter(hopped == 'Nanog')
Nanog.distinct <- Nanog.data %>%
  select(seqnames, start, end, strand) %>%
  distinct()
Nanog.gr <- makeGRangesFromDataFrame(Nanog.distinct, keep.extra.columns = T)
Nanog.distance <- as_tibble(distanceToNearest(Nanog.gr, donor, ignore.strand = T)) %>%
  mutate(
    location = start(Nanog.gr[queryHits]),
    upstream = location < start(donor),
    distance = ifelse(upstream == T, -distance, distance)
  )

# plot
ggplot(Nanog.distance, aes(x = location/1e6)) +
  geom_rect(data = LAD_left, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_rect(data = LAD_left_2, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_rect(data = LAD_right, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_histogram(fill = "gray85", color = "gray80",  binwidth = 0.1, alpha = 0.5)+
  geom_density(aes(y=..scaled..*69), n = 4096, size = 1) +
  scale_y_continuous(sec.axis = sec_axis(~./69, name = "Integration density", breaks = seq(0,1,0.2)))+
  geom_rug(length = unit(0.08, "npc")) +
  coord_cartesian(xlim = c(120.5, 125), ylim = c(0, 73), expand = FALSE) +
  theme_bw() +
  geom_vline(aes(xintercept = start(donor) / 1e6), col = RED, size = 1) +
  geom_vline(aes(xintercept = LAD_left$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_left$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_left_2$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_left_2$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_right$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_right$xmax), col = BLUE, size = 1) +

  labs(title = 'Nanog', x = "chr6 (Mb)", y = 'Integration counts') +
  theme(plot.title = element_text(hjust = 0.5),
        text = element_text(size = 50))


# save
# ggsave("SuppFig1E_LD20230919_Nanog.pdf", path = "./output/", height = 10, width = 20)

# number of integration
# total
nrow(Nanog.distance)

# in the plotted window
Nanog.distance %>% filter(location > 120.5e6 & location < 125e6) %>% write.table(file = "try", sep = "\t", quote=F, row.names = F)%>% nrow()


# number of integrations in iLAD
Nanog_ovl_integ <- findOverlaps(Nanog.gr, damid, select = "arbitrary")
Nanog.gr$damid <- damid$score[Nanog_ovl_integ]
Nanog.gr$LAD = ifelse(Nanog.gr$damid>0, "LAD", "iLAD")

Nanog.gr %>% data.frame() %>% filter(seqnames == "chr6") %>% filter(LAD == "iLAD") %>% nrow()

# number of integrations in iLAD in the plotted window
Nanog.gr %>% data.frame() %>% filter(seqnames == "chr6") %>% filter(start > 120.5e6 & start < 125e6) %>% filter(LAD == "iLAD") %>% nrow()



```
