---
title: "LD20230511_Figure1"
date: "2023-05-11"
output: 
  html_document:
    theme: journal #cerulean
    highlight: monochrome
    toc: true
    toc_float: true
    code_folding: show
  editor_options:
    chunk_output_type: console
---

Scripts for all figures in the paper.
This is Figure 1

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


# Load libraries
```{r}
suppressWarnings(library(GenomicRanges))
suppressWarnings(library(dplyr))
suppressWarnings(library(stringr))
suppressWarnings(library(readr))
suppressWarnings(library(tidyr))
suppressWarnings(library(ggplot2))
suppressWarnings(library(ggplotify))
library(rtracklayer)
library(IRanges)
library(GenomeInfoDb)
library(tidyverse)
library(ggrastr)
library(gridExtra)
library(colorspace)
library(caTools)
library(RColorBrewer)
library(ggpubr)

```


# Figure 1B - Tdgf1 locus with all insertions
```{r}
# data
## pA-DamID, CAST allele from my data
DamID <- as_tibble(import("/DATA/usr/l.dauban/projects/pA-DamID/E1970_ProkDNA/2nd_round/results_mm10_nextseq_strainspecific/tracks/normalized/bin-20kb/WT_F1_LB1_CAS-20kb-combined.bw")) %>% dplyr::rename(DamID = score) %>% filter(seqnames == "chr9")

## location of clones
clones.df = data.frame(sample = c('donor', 'T179', 'T44', 'T65', 'T162', 'T24', 'T48h20', 'T201h51', 'T201h143', 'T201h49', 'T3h153', 'T48h131', 'T48h81'),
           x = c(110927601, 111711810, 111912792, 112396323, 112440815, 112637307, 112877498, 112653616, 112360112, 112520473, 109078659, 109769723, 109627353),
           y = rep(-4, 13),
           xend = c(110927601, 111711810, 111912792, 112396323, 112440815, 112637307, 112877498, 112653616, 112360112, 112520473, 109078659, 109769723, 109627353),
           yend = rep(-3.5, 13))

## LAD location from HMM profiles at 20 kb
LAD_loc = data.frame(sample = c('small', 'big', 'other'),
           x = c(109420001, 111560001, 114140001),
           y = rep(1.5, 3),
           xend = c(109760000, 113640000, 114300000),
           yend = rep(1.5, 3))


# Function
PlotDataTracksLight <- function(input_tib, exp_names, color_groups, 
                                centromeres = NULL, plot_chr = "chr1", 
                                plot_start = 1, plot_end = 500e6,
                                smooth = 1, color_list = NULL,
                                fix_yaxis = F, aspect_ratio = NULL,
                                lighten_negative = F, raster = F) {
  
  # Get the scores for the samples
  tib_plot <- input_tib %>%
    dplyr::select(seqnames, start, end, 
                  all_of(exp_names))
  
  if (smooth > 1) {
    tib_plot <- tib_plot %>%
      mutate_at(vars(contains("_")), 
                runmean, k = smooth)
  }
  
  # Filter for plotting window
  tib_plot <- tib_plot %>%
    filter(seqnames == plot_chr,
           start >= plot_start,
           end <= plot_end)
  
  # Gather
  tib_gather <- tib_plot %>%
    gather(key, value, -seqnames, -start, -end) %>%
    mutate(key = factor(key, levels = exp_names),
           fill_column = color_groups[match(key, names(color_groups))],
           fill_column = factor(fill_column, levels = unique(color_groups)))
  
  # If desired, make negative values a lighter shade to improve visualization
  if (lighten_negative) {
    tib_gather <- tib_gather %>%
      mutate(fill_column = interaction(fill_column,
                                       value < 0))
  }
  
  
  
  # Plot
  ylimits <- quantile(tib_gather$value, c(0.001, 0.999), na.rm = T)
  fill_column <- NULL
  
  plt <- tib_gather %>% 
    ggplot(aes(fill = fill_column))
  
  
  # Set all counts tracks to the same limits
  if (raster) {
    plt <- plt + 
      rasterize(geom_rect(aes(xmin = start / 1e6, xmax = end / 1e6, 
                              ymin = 0, ymax = value)),
                dpi = 300)
  } else {
    plt <- plt + 
      geom_rect(aes(xmin = start / 1e6, xmax = end / 1e6, 
                    ymin = 0, ymax = value))
  }
  
  plt <- plt + 
    geom_hline(yintercept = 0, size = 0.5) +
    #geom_vline(xintercept = 110927601/1e6, size = 0.5, col = "grey") +
    #geom_vline(xintercept = 111912792/1e6, size = 0.5, col = "grey") +
    geom_segment(data=clones.df, mapping=aes(x=x/1e6, xend=xend/1e6, y=y, yend=yend), size =1)+
    geom_segment(data=LAD_loc, mapping=aes(x=x/1e6, xend=xend/1e6, y=y, yend=yend), size =1, color = "#6BAED6")+
    facet_grid(key ~ ., scales = "free_y") +
    xlab(paste0(plot_chr, " (Mb)")) +
    ylab("LaminB1/Dam only") +
    scale_x_continuous(expand = c(0, 0)) + 
    scale_y_continuous(expand = c(0.025, 0.025), limits = c(-4, 1.5)) +
    theme_classic() +
    theme(text = element_text(size = 50))
  
  # Centromeres
  if (! is.null(centromeres)) {
    centromeres.plt <- as_tibble(centromeres) %>%
      filter(seqnames == plot_chr) %>%
      gather(key_centromeres, value, start, end)
    
    plt <- plt +
      geom_line(data = centromeres.plt, 
                aes(x = value / 1e6, y = 0),
                color = "black", size = 2)
  }
  
  if (! is.null(color_list)) {
    
    if (lighten_negative) {
      color_list <- c(color_list,
                      lighten(color_list, amount = 0.5))
    }
    
    colors <- levels(tib_gather$fill_column)
    
    color_list <- color_list[1:length(colors)]
    names(color_list) <- colors
    #colors <- recode(colors, !!!color_list)
    
    plt <- plt +
      scale_fill_manual(values = color_list, guide = "none")
  } else {
    plt <- plt +
      scale_fill_brewer(palette = "Set1", guide = "none")
  }
  
  if (fix_yaxis) {
    plt <- plt + 
      coord_cartesian(xlim = c(max(c(plot_start,
                                     min(tib_gather$start))) / 1e6,
                               min(c(plot_end,
                                     max(tib_gather$end))) / 1e6),
                      ylim = ylimits)
  } else {
    plt <- plt + 
      coord_cartesian(xlim = c(max(c(plot_start,
                                     min(tib_gather$start))) / 1e6,
                               min(c(plot_end,
                                     max(tib_gather$end))) / 1e6))
  }
  
  if (! is.null(aspect_ratio)) {
    plt <- plt +
      theme(aspect.ratio = aspect_ratio)
  }
  
  plot(plt)
  
}

# Plot
PlotDataTracksLight(input_tib = DamID,
                    exp_names = c("DamID"),
                    color_groups = c(DamID = "pA-DamID"),
                    plot_chr = "chr9", plot_start = 108500000, plot_end = 119448158,
                    color_list = c("#A6CEE3", "#F5C374"),
                    #color_list = brewer.pal(3, "Paired"),
                    aspect_ratio = 1/3,
                    smooth = 4,
                    #fix_yaxis = F,
                    lighten_negative = T,
                    raster = T)

# save
ggsave("Fig1B_locus.pdf", height = 10, width = 20)


```


# Figure 1C - SB hopping behaviour in Tdgf1 iLAD
```{r}

# Load metadata of integrations
metadata <- dplyr::tibble(file = dir("/DATA/usr/l.dauban/projects/TagMap/E1682_SB_hopped_clones/Output/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::filter(grepl("insertions", file)) %>%
  dplyr::mutate(sample = stringr::str_remove(basename(file), "\\..*"))


# Load all mapped insertion sites into single tibble
tib <- dplyr::bind_rows(lapply(
  1:nrow(metadata),
  function(i) {
    tmp <- readr::read_tsv(metadata$file[i],
      col_types = cols(
        seqnames = col_character(),
        start = col_double(),
        end = col_double(),
        start_gap = col_double(),
        end_gap = col_double(),
        gap_concordance = col_double(),
        count = col_double(),
        D = col_double(),
        p = col_double(),
        concordance = col_double(),
        p_adj = col_double(),
        strand = col_character(),
        name = col_character(),
      )
    ) %>%
      dplyr::mutate(sample = metadata$sample[i])
  }
))

# Convert 'seqnames' to factor
chromosomes <- paste0("chr", c(1:19, "X"))
tib <- tib %>%
  dplyr::mutate(seqnames = factor(seqnames, levels = chromosomes)) %>%
  tidyr::drop_na(seqnames)

# Add factor 'hopped' describing source of hopping
tib <- tib %>%
  dplyr::mutate(sample = factor(sample, )) %>%
  dplyr::mutate(
    hopped = case_when(
      grepl("Fzd5", sample) ~ "Fzd5",
      grepl("Tdgf1", sample) ~ "Tdgf1",
      grepl("Nanog", sample) ~ "Nanog",
      T ~ "unknown"
    ),
    hopped = factor(hopped, c("Fzd5", "Tdgf1", "Nanog"))
  ) %>%
  # Add factor 'library' describing samples originating from pool or clone libraries
  dplyr::mutate(
    library = case_when(
      grepl("_D", sample) ~ "pool",
      grepl("_cl", sample) ~ "clone",
      grepl("_FACS", sample) ~ "pool",
      T ~ "unknown"
    ),
    library = factor(library, c("pool", "clone"))
  )


# data
donor <- GenomicRanges::GRanges(seqnames = "chr9", ranges = IRanges(start = 110927601, end = 110927601), strand = "*")

LAD_small <- data.frame(xmin=109420001/1e6, xmax=109760000/1e6, ymin=-Inf, ymax=Inf)
LAD_big <- data.frame(xmin=111560001/1e6, xmax=113640000/1e6, ymin=-Inf, ymax=Inf)

# Color-blind colors
BLUE <- '#56B4E9'
YELLOW <- '#F0E442'
RED <- '#D55E00'

# process
Tdgf1.data <- tib %>%
  filter(hopped == 'Tdgf1')
Tdgf1.distinct <- Tdgf1.data %>%
  select(seqnames, start, end, strand) %>%
  distinct()
Tdgf1.gr <- makeGRangesFromDataFrame(Tdgf1.distinct, keep.extra.columns = T)
Tdgf1.distance <- as_tibble(distanceToNearest(Tdgf1.gr, donor, ignore.strand = T)) %>%
  mutate(
    location = start(Tdgf1.gr[queryHits]),
    upstream = location < start(donor),
    distance = ifelse(upstream == T, -distance, distance)
  )

# select
Tdgf1.distance_subset = Tdgf1.distance %>% filter(location>109e6 & location<114e6)

# plot
## version 2 - without
ggplot(Tdgf1.distance_subset, aes(x = location/1e6)) +
  geom_rect(data = LAD_small, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_rect(data = LAD_big, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_histogram(fill = "gray85", color = "gray80",  binwidth = 0.1, alpha = 0.5)+
  geom_density(aes(y=..scaled..*43), n = 4096, size = 1, adjust = 8) +
  scale_y_continuous(sec.axis = sec_axis(~./43, name = "Integration density", breaks = seq(0,1,0.2)))+
  geom_rug(length = unit(0.08, "npc")) +
  coord_cartesian(xlim = c(109e6, 114e6)/1e6, ylim = c(0, 45), expand = FALSE) +
  theme_bw() +
  geom_vline(aes(xintercept = start(donor) / 1e6), col = RED, size = 1) +
  geom_vline(aes(xintercept = LAD_small$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_small$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_big$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_big$xmax), col = BLUE, size = 1) +
  labs(title = 'Tdgf1', x = "chr9 (Mb)", y = 'Integration counts') +
  theme(plot.title = element_text(hjust = 0.5),
        text = element_text(size = 50)) 


# save
ggsave("Fig1C_LD20230919b_Hopping_iLAD.pdf", height = 10, width = 20)

# number of integration
nrow(Tdgf1.distance)

# number of integrations within 2 Mb
Tdgf1.distance %>% filter(distance<1e6)

# in the plotted window
Tdgf1.distance_subset %>% filter(location > 109e6 & location < 114e6) %>% nrow()

```


# Figure 1Da - SB hopping behaviour in Tdgf1 big LAD
```{r}
############# Combine data from E1792 and E1828

# Load metadata of all files
metadata <- dplyr::tibble(file = dir("/DATA/usr/k.rademaker/projects/E1792/Output/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::add_row(file = dir("/DATA/usr/l.dauban/projects/TagMap/E1828_2nd_round_T3_T48_T201/KR_pipeline/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::add_row(file = dir("/DATA/usr/l.dauban/projects/TagMap/E1828_2nd_round_T3_T48_T201/KR_pipeline/oneread/mm10/home/insertions/", recursive = T, full.names = T)) %>%
  dplyr::filter(grepl("insertions", file)) %>%
  dplyr::mutate(sample = stringr::str_remove(basename(file), "\\..*"))

# Load all mapped insertion sites into single tibble
tib <- dplyr::bind_rows(lapply(
  1:nrow(metadata),
  function(i) {
    tmp <- readr::read_tsv(metadata$file[i],
      col_types = cols(
        chr = col_character(),
        start = col_double(),
        end = col_double(),
        seq = col_character(),
        region_start = col_double(),
        region_end = col_double(),
        read_count = col_double(),
        mapq = col_double(),
        read_count_1 = col_double(),
        mapq_1 = col_double(),
        read_count_2 = col_double(),
        mapq_2 = col_double(),
      )
    ) %>%
      dplyr::mutate(sample = metadata$sample[i])
  }
))

# Convert 'chr' to factor
chromosomes <- paste0("chr", c(1:19, "X"))
tib <- tib %>%
  dplyr::mutate(chr = factor(chr, levels = chromosomes)) %>%
  tidyr::drop_na(chr)

# Add factor 'hopped' describing source of hopping
tib <- tib %>% dplyr::mutate(sample = factor(sample, )) %>%
  dplyr::mutate(
    hopped = case_when(
      grepl("N37_", sample) ~ "N37",
      grepl("N31_", sample) ~ "N31",
      grepl("F108_", sample) ~ "F108",
      grepl("T201_", sample) ~ "T201",
      grepl("T137_", sample) ~ "T137",
      grepl("T48_", sample) ~ "T48",
      grepl("T3_", sample) ~ "T3",
      grepl("Tdgf1_", sample) ~ "Tdgf1",
      grepl("Fzd5_", sample) ~ "Fzd5",
      T ~ "unknown"
    ),
    hopped = factor(hopped, c("N37", "N31", "F108", "T201", "T137", "T48", "T3", "Tdgf1", "Fzd5")))


# filter
tib_T201 <- tib %>% filter(hopped == 'T201')


# Distance of integrations to hopping locations
tib_T201_distinct <- tib_T201 %>%
  dplyr::select(chr, start, end, strand) %>%
  dplyr::distinct()

T201 <- GenomicRanges::GRanges(seqnames = "chr9", ranges = IRanges(start = 112440264), strand = "*")

gr_T201 <- GenomicRanges::makeGRangesFromDataFrame(tib_T201_distinct, keep.extra.columns = T)
T201_distance <- dplyr::as_tibble(GenomicRanges::distanceToNearest(gr_T201, T201, ignore.strand = T)) %>%
  dplyr::mutate(location = start(gr_T201[queryHits]),
                upstream = location < start(T201),
                distance = ifelse(upstream == T, -distance, distance)) %>%
  dplyr::mutate(distance = start(T201) + distance)
    

# select
T201_distance_subset = T201_distance %>% filter(distance>110.5e6&distance<114.5e6)

# plot
##version 1 - with subsetting
ggplot(T201_distance_subset, aes(x = distance / 1e6)) +
  geom_rect(data = LAD_big, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_histogram(fill = "gray85", color = "gray80",  binwidth = 0.1, alpha = 0.5)+
  geom_density(aes(y=..scaled..*51), n = 4096, size = 1, adjust = 8) +
  scale_y_continuous(sec.axis = sec_axis(~./51, name = "Integration density", breaks = seq(0,1,0.2)))+
  geom_rug(length = unit(0.08, "npc")) +
  geom_hline(yintercept = 0, col = 'black', size = 0.5) +
  coord_cartesian(xlim = c(110.5, 114.5), ylim = c(0, 55), expand = FALSE) +
  theme_bw() +
  geom_vline(aes(xintercept = start(T201) / 1e6), col = RED, size = 1) +
  #geom_vline(aes(xintercept = LAD_small$xmin), col = BLUE, size = 1) +
  #geom_vline(aes(xintercept = LAD_small$xmax), col = BLUE, size = 1) +
  #geom_rect(data = LAD_small, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_vline(aes(xintercept = LAD_big$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_big$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = 84687482 / 1e6), col = RED, size = 1) +
  labs(title = 'Tdgf1 clone T201', x = "chr9 (Mb)", y = 'Integration counts') +
  scale_x_continuous(breaks = seq(110, 115, 1),
                     labels = c("110", "111", "112", "113", "114", "115")) +
  theme(plot.title = element_text(hjust = 0.5),
        text = element_text(size = 50))


# save
ggsave("Fig1D_LD20230919b_Hopping_bigLAD2.pdf", height = 10, width = 20)


# number of integration
# total
nrow(T201_distance_subset)

# in the plotted window
T201_distance_subset %>% filter(location > 110.5e6 & location < 114.5e6) %>% nrow()


# number of integrations in iLAD
damid <- import("/DATA/usr/l.dauban/projects/pA-DamID/E1970_ProkDNA/2nd_round/results_mm10_nextseq_strainspecific/tracks/normalized/bin-20kb/WT_F1_LB1_CAS-20kb-combined.bw")

T201_distance_subset.gr = gr_T201 %>% data.frame() %>% filter(seqnames == "chr9" & start>start(T201)-26e6 & start<start(T201)+26e6) %>% GRanges()
T201_distance_subset_ovl_integ <- findOverlaps(T201_distance_subset.gr, damid, select = "arbitrary")
T201_distance_subset.gr$damid <- damid$score[T201_distance_subset_ovl_integ]
T201_distance_subset.gr$LAD = ifelse(T201_distance_subset.gr$damid>0, "LAD", "iLAD")

T201_distance_subset.gr %>% data.frame() %>% filter(LAD == "iLAD")


# number of integrations within LAD2
T201_distance_subset.gr %>% data.frame() %>% filter(seqnames == "chr9" & start>LAD_big$xmin*1e6 & start<LAD_big$xmax*1e6)


```


```{r}
# 
# 
# ## location of clones
# clones.df = data.frame(sample = c('T179', 'T44', 'T65', 'T162', 'T24', 'T48h20', 'T201h51', 'T201h143', 'T201h49', 'T3h153', 'T48h131', 'T48h81'),
#            x = c(111711810, 111912792, 112396323, 112440815, 112637307, 112877498, 112653616, 112360112, 112520473, 109078659, 109769723, 109627353),
#            y = rep(-1, 12),
#            xend = c(111711810, 111912792, 112396323, 112440815, 112637307, 112877498, 112653616, 112360112, 112520473, 109078659, 109769723, 109627353),
#            yend = rep(1, 12),
#            rec = c(rep("del", 6), rep("inv", 6)),
#            col = c(rep("#733B63", 6), rep("#F27289", 6)))
# 
# 
# 
# ggplot(T201_distance_subset, aes(x = distance / 1e6, color = rec)) +
#     geom_segment(data=clones.df, mapping=aes(x=x/1e6, xend=xend/1e6, y=y, yend=yend), size =1)+
#   geom_hline(yintercept = 0, col = 'black', size = 0.5) +
#   coord_cartesian(xlim = c(110.5e6, 114.5e6) / 1e6, expand = FALSE) +
#   theme_bw() +
#   geom_vline(aes(xintercept = LAD_small$xmin), col = BLUE, size = 1) +
#   geom_vline(aes(xintercept = LAD_small$xmax), col = BLUE, size = 1) +
#   geom_rect(data = LAD_small, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
#   geom_vline(aes(xintercept = LAD_big$xmin), col = BLUE, size = 1) +
#   geom_vline(aes(xintercept = LAD_big$xmax), col = BLUE, size = 1) +
#   geom_rect(data = LAD_big, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
#   geom_vline(aes(xintercept = 84687482 / 1e6), col = RED, size = 1) +
#   labs(title = 'Tdgf1 clone T201', x = "chr9 (Mb)", y = 'Integration density') +
#   scale_x_continuous(breaks = seq(110, 115, 1),
#                      labels = c("110", "111", "112", "113", "114", "115")) +
#   scale_color_manual(values = c("#733B63", "#F27289")) +
#   theme(plot.title = element_text(hjust = 0.5),
#         text = element_text(size = 50))
# 

```


# Figure 1Db - SB hopping behaviour in Tdgf1 small LAD
```{r}

T48 <- GenomicRanges::GRanges(seqnames = "chr9", ranges = IRanges(start = 109626922, end = 109626922), strand = "*")

# filter
tib_T48 <- tib %>% filter(hopped == 'T48')


# Distance of integrations to hopping locations
tib_T48_distinct <- tib_T48 %>%
  dplyr::select(chr, start, end, strand) %>%
  dplyr::distinct()

gr_T48 <- GenomicRanges::makeGRangesFromDataFrame(tib_T48_distinct, keep.extra.columns = T)
T48_distance <- dplyr::as_tibble(GenomicRanges::distanceToNearest(gr_T48, T48, ignore.strand = T)) %>%
  dplyr::mutate(location = start(gr_T48[queryHits]),
                upstream = location < start(T48),
                distance = ifelse(upstream == T, -distance, distance)) %>%
  dplyr::mutate(distance = start(T48) + distance)
    

# # select
T48_distance_subset = T48_distance %>% filter(location > 107.5e6 & location<111.5e6)

# plot
ggplot(T48_distance_subset, aes(x = distance / 1e6)) +
  #geom_rect(data = LAD_big, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_rect(data = LAD_small, aes(xmin = xmin, xmax = xmax, ymin = ymin, ymax = ymax), fill = "#A6CEE3", alpha = 0.5, inherit.aes = FALSE) +
  geom_histogram(fill = "gray85", color = "gray80",  binwidth = 0.1, alpha = 0.5)+
  geom_density(aes(y=..scaled..*40), n = 4096, size = 1, adjust = 8) +
  scale_y_continuous(sec.axis = sec_axis(~./40, name = "Integration density", breaks = seq(0,1,0.2)))+
  geom_rug(length = unit(0.08, "npc")) +
  geom_hline(yintercept = 0, col = 'black', size = 0.5) +
  coord_cartesian(xlim = c(107.5, 111.5), ylim = c(0, 43), expand = FALSE) +
  theme_bw() +
  geom_vline(aes(xintercept = start(T48) / 1e6), col = RED, size = 1) +
  geom_vline(aes(xintercept = LAD_small$xmin), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = LAD_small$xmax), col = BLUE, size = 1) +
  #geom_vline(aes(xintercept = LAD_big$xmin), col = BLUE, size = 1) +
  #geom_vline(aes(xintercept = LAD_big$xmax), col = BLUE, size = 1) +
  geom_vline(aes(xintercept = 84687482 / 1e6), col = RED, size = 1) +
  scale_x_continuous(breaks = seq(107, 112, 1),
                     labels = c("107", "108", "109", "110", "111", "112")) +
  labs(title = 'Tdgf1 clone T48', x = "chr9 (Mb)", y = 'Integration counts') +
  theme(plot.title = element_text(hjust = 0.5),
        text = element_text(size = 50))

# save
ggsave("Fig1D_LD20230919b_Hopping_smallLAD2.pdf", height = 10, width = 20)


# number of integration
nrow(T48_distance)

# total
nrow(T48_distance_subset)

# in the plotted window
T48_distance_subset %>% filter(location > 107.5e6 & location < 111.5e6) %>% nrow()


# number of integrations in iLAD
T48_ovl_integ <- findOverlaps(gr_T48, damid, select = "arbitrary")
gr_T48$damid <- damid$score[T48_ovl_integ]
gr_T48$LAD = ifelse(gr_T48$damid>0, "LAD", "iLAD")

gr_T48 %>% data.frame() %>% filter(seqnames == "chr9") %>% filter(LAD == "iLAD")

# number of integrations within LAD2
gr_T48 %>% data.frame() %>% filter(seqnames == "chr9" & start>LAD_small$xmin*1e6 & start<LAD_small$xmax*1e6)


```


# Figure 1E - Cre recombination
```{r}
rec_eff_table <- read.table("/DATA/usr/l.dauban/projects/Paper/rec_eff_graph.txt", header = TRUE)

rec_eff_table %>% filter(Location == "T201h49" | Location == "T201.h179" | Location == "T201.h44" |  Location == "T201.h65" | Location == "T201.h162" | Location == "T201.h24" | Location == "T48.h20" | Location == "T48.h81" | Location == "T3.h153" | Location == "T201h51" | Location == "T201h143" | Location == "T3") %>% ggplot(aes(x=Distance, y=Rec_eff, color = Rec_type))+
  geom_point(aes(size = 4)) +
  labs(x = "Distance between loxP sites (Mb)", y = "Cre recombination efficiency (%)", color = "Recombination type") +
  coord_cartesian(x = c(0, 2), y = c(0, 23)) +
  #scale_color_brewer(palette="PuRd", breaks = 5:8) +
  scale_color_manual(values = c("#733B63", "#F27289")) +
  theme_bw() +
  theme(text = element_text(size = 20)) +
  guides(size = "none")


ggsave("Fig1E_LD20230718_rec_eff.pdf", width = 10, height = 6)


rec_eff_table %>% filter(Location == "T201h49" | Location == "T201.h179" | Location == "T201.h44" |  Location == "T201.h65" | Location == "T201.h162" | Location == "T201.h24" | Location == "T48.h20" | Location == "T48.h81" | Location == "T3.h153" | Location == "T201h51" | Location == "T201h143" | Location == "T3") %>% 
  ggplot(aes(x=Distance, y=Rec_eff))+
  geom_point(aes(size = 4)) +
  labs(x = "Distance between loxP sites (Mb)", y = "Cre recombination efficiency (%)", color = "Recombination type") +
  coord_cartesian(x = c(0, 2), y = c(0, 23)) +
  #scale_color_brewer(palette="PuRd", breaks = 5:8) +
  #scale_color_manual(values = c("#733B63", "#F27289")) +
  theme_bw() +
  theme(text = element_text(size = 20)) +
  geom_smooth(method=lm) +
  stat_cor(method = "pearson") +
  guides(size = "none")



```


